### PR TITLE
Cleanup Referrals

### PR DESCRIPTION
**Pull request by tiffwong**
## Description

Removal conditional logic since the referral changes have been released to GA

## How to test

Make sure the referral states look like they're supposed to, message me if you need help seeing the different states

### CODE REVIEW

File: packages/manager/src/featureFlags.ts [hunk 1]
```diff
@@ -22,7 +22,6 @@ export interface Flags {
   firewalls: boolean;
   oneClickApps: OneClickApp;
   promotionalOffers: PromotionalOffer[];
-  thirdPartyAuth: boolean;
```
Comment (tiffwong) [inline packages/manager/src/featureFlags.ts:25]: This flag is also no longer used so I removed it

File: packages/manager/src/features/Profile/Referrals/Referrals.tsx [hunk 2]
```diff
@@ -103,183 +91,111 @@ type CombinedProps = StateProps;
...
-    code,
-    url,
-    total,
-    completed,
-    pending,
-    credit,
-  } = props;
+  const { profileLoading, url, total, completed, pending, credit } = props;
 
   const allowReferral = url !== undefined;
```
Comment (bnussman) [inline packages/manager/src/features/Profile/Referrals/Referrals.tsx:97]: Getting an empty field. Do we need this change again?
```ts
const allowReferral = Boolean(url);
```

**Review by bnussman — CHANGES_REQUESTED** [review]

File: packages/manager/src/features/Profile/Referrals/Referrals.tsx (hunk 2 above)
Comment (DevDW) [reply packages/manager/src/features/Profile/Referrals/Referrals.tsx:97]: I'm getting the same thing -- I think that when the hotfix went out for this, `staging` wasn't merged into `develop` so this change didn't get in

Comment (tiffwong) [reply packages/manager/src/features/Profile/Referrals/Referrals.tsx:97]: @bnussman @DevDW that's exactly what happened...thanks for catching that! 😓

**Review by DevDW — APPROVED** [review]
LGTM now, can you add the other changes from https://github.com/linode/manager/pull/7732/files so that `develop` gets updated when you merge this?

**Review by bnussman — APPROVED** [review]

**Pull request merged**